NBitcoin/BouncyCastle/crypto/signers/ECDsaSigner.cs

```diff
@@ -99,16 +103,22 @@ public virtual BigInteger[] GenerateSignature(byte[] message)
 			do // Generate s
 			{
 				BigInteger k;
+				byte[] rBytes = null;
 				do // Generate r
 				{
-					k = kCalculator.NextK();
+					do 
+					{
+						k = kCalculator.NextK();
 
-					ECPoint p = basePointMultiplier.Multiply(ec.G, k).Normalize();
+						ECPoint p = basePointMultiplier.Multiply(ec.G, k).Normalize();
 
-					// 5.3.3
-					r = p.AffineXCoord.ToBigInteger().Mod(n);
+						// 5.3.3
+						r = p.AffineXCoord.ToBigInteger().Mod(n);
+					}
+					while(r.SignValue == 0);
+					rBytes = r.ToByteArrayUnsigned();
 				}
-				while(r.SignValue == 0);
+				while(forceLowR && rBytes[0] >= 0x80);
```

Comment:
nit: Replace `rBytes[0]` by `r.ToByteArrayUnsigned()[0]` so the array is created only when strictly needed, and you dont' need to declare `rBytes` above.